src/StdStyle.sol

```diff
@@ -0,0 +1,241 @@
+// SPDX-License-Identifier: MIT
+pragma solidity >=0.4.22 <0.9.0;
+
+import {Vm} from "./Vm.sol";
+
+library StdStyle {
+    Vm private constant vm = Vm(address(uint160(uint256(keccak256("hevm cheat code")))));
+
+    string constant RED = "\u001b[91m";
+    string constant GREEN = "\u001b[92m";
+    string constant YELLOW = "\u001b[93m";
+    string constant BLUE = "\u001b[94m";
+    string constant MAGENTA = "\u001b[95m";
+    string constant CYAN = "\u001b[96m";
+    string constant BOLD = "\u001b[1m";
+    string constant DIM = "\u001b[2m";
+    string constant ITALIC = "\u001b[3m";
+    string constant UNDERLINE = "\u001b[4m";
+    string constant REVERSE = "\u001b[7m";
+    string constant RESET = "\u001b[0m";
+
+    function red(string memory self) internal view returns (string memory) {
+        return string.concat(RED, self, RESET);
+    }
+
+    function red(uint256 self) internal view returns (string memory) {
+        return red(vm.toString(self));
+    }
+
+    function red(int256 self) internal view returns (string memory) {
+        return red(vm.toString(self));
+    }
+
+    function red(address self) internal view returns (string memory) {
+        return red(vm.toString(self));
+    }
+
+    function red(bool self) internal view returns (string memory) {
```

Comment:
Seems we're missing `bytes32` and `bytes memory` overloads for each color/format, could we add those?

Reply:
This is intentional, I originally added `bytes32` and `bytes memory` too. However, if you then try to pass as string as literal i.e: `red("Test");` the compiler is not able to differentiate which signature to use and prints an error stating there are multiple candidates. Solution would be to use `red(string("Test"));` but I think this will be more tedious as user will use more often a string than bytes types + users can do `red(vm.toString(bytes32));`

However, other solution to overcome this could be `redBytes32()`and `redBytes()` wdty @mds1?

Reply:
Ah true, let's go with `redBytes32` and `redBytes` for each, since that matches e.g. `console.logBytes` and `console.logBytes32`

Reply:
Added `Bytes` and `Bytes32` types for each style